sendQueued.js

```diff
@@ -1,130 +1,158 @@
-            }, function(err, result) {
-              if (err) {
-                return console.log("client.sendMessage", err);
-              }
-
-              console.log('Queued message sent to ' + phone);
-
-              knex('queued')
-                  .where('queued_id', '=', queuedCitation.queued_id)
-                  .update({'sent': true,
-                            'asked_reminder': true,
-                            'asked_reminder_at' : new Date()})
-                  .exec(function (err, results) {
-                    if (err) {
-                      console.log(err);
-                    }
+/**
+ * Find a citation that is related to a queued message.
+ * 
+ * @param  {Object} queuedMessage for which we want to lookup citation data.
+ * @return {Promise}  promise to retrieve citation data.
+ */
+function retrieveCitation(queuedMessage) {
+  return new Promise(function(resolve, reject) {
+    db.findCitation(queuedMessage.citation_id, function(err, results) {
+      resolve({
+        queuedMessage: queuedMessage,
+        citationFound: results.length > 0,
+        relatedCitation: (results.length ? results[0] : false)
+      });
+    });
+  });
+};
 
-                    count++;
+/**
+ * Process citation:
+ *   1.)  Citation data found:  send message to defendant asking if they want a reminder
+ *   2.)  Citation data not found and message has been queued too long:  send a "not found" message to defendant.
+ *   3.)  N/A do nothing and leave queued.
+ * 
+ * @param  {Object} queued queued message and citation data(if found)
+ * @return {Promise} promise to process queued message (if applicable)
+ */
+function processCitationMessage(queued) {
+  return new Promise(function(resolve, reject) {
+    var decipher = crypto.createDecipher('aes256', process.env.PHONE_ENCRYPTION_KEY),
+        phone = decipher.update(queued.queuedMessage.phone, 'hex', 'utf8') + decipher.final('utf8');
 
-                    if (count === queued.length) {
-                      resolve();
-                    }
-                  });
-            });
-          });
-        } else {
-          var daysSinceCreation = moment().diff(moment(queuedCitation.created_at), 'days');
+    if (queued.citationFound) {
+      var name = strings.scrubName(queued.relatedCitation.defendant),
+          datetime = dates.fromDateAndTime(queued.relatedCitation.date, queued.relatedCitation.time);
 
-          if (daysSinceCreation > parseInt(process.env.QUEUE_TTL_DAYS)) {
-            knex('queued')
-              .where('queued_id', '=', queuedCitation.queued_id)
-              .update({'sent': true})
-              .exec(function(err, results) {
-                if (err) {
-                  console.log(err);
-                }
-              });
+      Promise.resolve()
+        .then(sendMessage(phone, process.env.TWILIO_PHONE_NUMBER, messages.greetingOneOfTwo()))
```

Comment:
Last night, we changed this to only send one message. Twilio will handle the pagination for us.